### ISSUE TITLE
Make large world map objects (which have a bigger hitbox than its graphical representation) being always visible

### ISSUE THREAD

**Issue by Wohlstand** [Wohlstand/TheXTech#446]
There is a problem that objects with a bigger graphical representation get not drawn until their hitbox appears on the visible scene. To solve it, I need to index world map objects by their graphical body and show them when their graphical body will appear on the screen.

**Comment by SkullGamer205** [comment]
So cool.
For 1.3.6.1?

**Comment by ds-sloth** [comment]
This should be a pretty quick feature to implement.

Which objects are we trying to add it for? Tiles, paths, scenes, and levels?

Do we want the user to specify the increased size somehow, or do we assume that the frame count is the same and guess the new size based on the graphics?

The easiest way to implement would be by changing the `Location` field of the object, but this would change the logic and require a compat field. Or, we could do it safely by modifying the lookup table location extraction code for world map objects to use the graphical width / height, and rewriting the draw code only.

**Comment by ds-sloth** [comment]
I believe that this issue only affects levels with `GFXLevelBig` set. Is that correct? If so, the commit I just pushed should resolve this issue.

I also have a version targeted at 1.3.6.x that specializes `Tree_Extractor` for `WorldLevelRef_t` (add the following block to `trees.cpp`). That's the only conflict in the cherry-pick.

```
template<>
class Tree_Extractor<WorldLevelRef_t>
{
public:
    static void ExtractBoundingBox(const WorldLevelRef_t object, loose_quadtree::BoundingBox<double> *bbox)
    {
        Location_t tempLocation = object->LocationGFX();

        bbox->left      = tempLocation.X;
        bbox->top       = tempLocation.Y;
        bbox->width     = tempLocation.Width;
        bbox->height    = tempLocation.Height;
    }
};
```

**Comment by ds-sloth** [comment]
Can somebody please test this case and confirm whether it has already been fixed? There is no need to separately backport it because the main branch will be used for release 1.3.6.1.

**Comment by SkullGamer205** [comment]
This didn't work...
Screenshots below..

https://cdn.discordapp.com/attachments/692079250215403624/1128633700578500738/image.png
https://cdn.discordapp.com/attachments/692079250215403624/1128633700851142776/image.png
https://cdn.discordapp.com/attachments/692079250215403624/1128633701102796800/image.png

`main` branch
version: latest (git)

**Comment by ds-sloth** [comment]
Thanks, but that is a test for a different (related) feature.

This issue is specific to the objects with large vanilla GFX (and small vanilla hitbox), and only appears when they are near the edge of the viewport.

**Comment by ds-sloth** [comment]
This has been fixed for a long time (in the changelog of 1.3.6.1). I quickly added some code to also make sure that "Big Paths" behind levels are drawn correctly. Closing now.
